packages/components/src/components/domainproperties/dataset/DatasetPropertiesPanelFormElements.tsx

```diff
@@ -0,0 +1,297 @@
+    onCategoryChange?: any;
+    label?: string;
+}
+
+interface BasicPropertiesTitleProps {
+    title: string
+}
+
+export class BasicPropertiesTitle extends React.PureComponent<BasicPropertiesTitleProps> {
+    render() {
+        return <div className="domain-field-section-heading">{this.props.title}</div>;
+    }
+}
+
+export class DescriptionInput extends React.PureComponent<BasicPropertiesInputsProps> {
+    render() {
+        const { description, onInputChange } = this.props;
+        const value = description === null ? '' : description;
+
+        return(
+            <Row className={'margin-top'}>
+                <Col xs={4}>
+                    Description
+                </Col>
+
+                <Col xs={7}>
+                    <textarea
+                        className="form-control textarea-noresize"
+                        id="description"
+                        value={value}
+                        onChange={onInputChange}
+                    />
+                </Col>
+            </Row>
+        );
+    }
+}
+
+export class SelectPropertyInput extends React.PureComponent<any> {
+    render() {
+        const { label, labelKey, valueKey, onInputChange, options, disabled } = this.props;
+
+        return(
+            <Row className={'margin-top'}>
+                <Col xs={4} >
+                    {label}
+                </Col>
+
+                <Col xs={7} >
+                    <SelectInput
+                        onChange={onInputChange}
+                        options={options}
+                        inputClass="" // This attr is necessary for proper styling
+                        labelClass=""
+                        valueKey={valueKey}
+                        labelKey={labelKey}
+                        formsy={false}
+                        multiple={false}
+                        required={false}
+                        disabled={disabled}
+                    />
+                </Col>
+
+                <Col lg={3}/>
+            </Row>
+        );
+    }
+}
+
+export class BasicPropertiesFields extends React.PureComponent<BasicPropertiesInputsProps, any> {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+             category: this.props.model.categoryId,
+             availableCategories: []
+        };
+    }
+
+    componentDidMount() {
+        const { model } = this.props;
+
+        // Ajax call handling to get available categories
+        fetchCategories()
+            .then((data) => {
+                this.setState(() => ({
+                    availableCategories: data.categories
+                }))
+            })
+    }
+
+    render() {
+        const { name, description, categoryId, label, onInputChange, onCategoryChange } = this.props;
+        const { availableCategories } = this.state;
+
+        const nameTip =
+            <>
+                Required. This name must be unique. It is used when identifying datasets during data upload.
```

Comment:
Same comment as before, take out the "Required." text here and use the required prop via <DomainFieldLabel>.